HACKING.md

```diff
@@ -42,73 +53,135 @@ Add `$GOPATH/bin` to your `PATH`, so you can run the go programs you install:
 your `$GOPATH` is more complex than a single entry you'll need to adjust the
 above).
 
-Note that if you are using go 1.16 or newer you need to disable the
-go modules feature. Use:
 
-    export GO111MODULE=off
+### Getting the snapd sources
 
-for this.
+The easiest way to get the source for `snapd` is to clone the Github repository in
+a directory where you have read-write permissions, such as your home directory.
 
+    cd ~/
+    git clone https://github.com/snapcore/snapd.git
+    cd snapd
 
-### Getting the snapd sources
+This will allow you to build and test `snapd`. If you wish to contribute to the
+`snapd` project, please see the Contributing section.
+
+### Contributing
+
+Contributions are always welcome!
+
+Please make sure that you sign the Canonical contributor agreement [here](
+http://www.ubuntu.com/legal/contributors).
 
-The easiest way to get the source for `snapd` is to use the `go get` command.
+Complete requirements can be found in CONTRIBUTING.md.
 
-    go get -d -v github.com/snapcore/snapd/...
+Before you contribute to the`snapd` project, please fork the `snapd` repository
+(under your Github account), and submit patches through a Pull Request.
 
-This command will checkout the source of `snapd` and inspect it for any unmet
-Go package dependencies, downloading those as well. `go get` will also build
-and install `snapd` and its dependencies. To also build and install `snapd`
-itself into `$GOPATH/bin`, omit the `-d` flag. More details on the `go get`
-flags are available using
+```
+cd ~/
+git clone https://github.com/<user>/snapd.git
```

Comment:
I'm not sure that this is a good suggestion. I'd rather have people fork our main repo, and add their own fork as a remote, otherwise (unless they are one-time contributors) their repo will get our of sync.

Also, did you encounter any problems with `go get`? It has the advantage that it downloads the repo in the proper location.